### ISSUE TITLE
Automatically publish development orb version every x days

### ISSUE THREAD

**Issue by gmemstr** [CircleCI-Public/Orb-Project-Template#9]
## Describe Request:

Similar to https://github.com/CircleCI-Public/orb-starter-kit/issues/44 - occasionally, orb `dev:alpha` versions will expire, which means that future commits for that orb will fail the pipeline. Ideally, we set up a cron job within the configuration to automatically refresh this version every 90 (or 89?) days.

**Comment by SimonBarendse** [comment]
We've done the same thing as you suggest in https://github.com/secrethub/secrethub-circleci-orb/pull/39. Shall I submit a PR that adds the `keep-dev-orb` from that PR?

**Comment by daniel-shuy** [comment]
@SimonBarendse Yes please, I'm facing this issue as well, hopefully it will help others from facing it.

Meanwhile I'll copy the code from your PR, thanks!

**Comment by daniel-shuy** [comment]
But now that I think about it, I highly doubt CircleCI will merge the PR since they probably expired dev orbs for a reason 😅

**Comment by KyleTryon** [comment]
Hey folks, we are currently working on the next iteration of Orb Tools (v11) and everything that goes with this, meaning this project here the Orb Project Template, and documentation. The update should be applied by the end of July.

In Orb tools 11, this will not be an issue _and_ we won't need to use scheduled jobs either. Orb tools 11 will make use of CircleCI's new [Dynamic Config](https://circleci.com/docs/2.0/dynamic-config/) to solve this.

The problem today, is the config file requires the dev orb (which only last 90 days), but it only needs this orb for the second workflow where there are integration tests. With Dynamic Config in orb tools 11, we will have 2 distinct config files, an initial "setup" config which contains our first workflow which does linting, shellchecking, unit testing, and finally publishes the dev version of the orb before continuing on to the second config file. The second config file is where the dev orb is required for integration testing.

With this new design, the config will never call on an orb that does not exist.

This work is still very early and there are many core changes planned that are not yet present here but this will give you an idea of the new config design if you would like a sneak peek: https://github.com/CircleCI-Public/orb-tools-orb/pull/111

**Comment by netcazean** [comment]
#31
